src/scheduler/misc.c

```diff
@@ -814,22 +814,32 @@ calc_time_left_STF(resource_resv *resresv, sch_resource_t* min_time_left)
  *		calc_time_left - calculate the remaining time of a resource resv
  *
  * @param[in]	resresv	-	the resource resv to calculate
+ * @param[in]	use_hard_duration - use the resresv's hard_duration instead of normal duration
  *
  * @return	time left on job
  * @retval	-1	: on error
  *
  */
 int
-calc_time_left(resource_resv *resresv)
+calc_time_left(resource_resv *resresv, int use_hard_duration)
 {
 	time_t used_amount = 0;
+	long duration;
 
-	if (resresv->duration == UNSPECIFIED)
+	if(use_hard_duration && resresv->hard_duration == UNSPECIFIED)
 		return -1;
 
+	else if (!use_hard_duration && resresv->duration == UNSPECIFIED)
+		return -1;
+
+	if(use_hard_duration)
+		duration = resresv->hard_duration;
+	else
+		duration = resresv->duration;
+
 	used_amount = calc_used_walltime(resresv);
 
-	return IF_NEG_THEN_ZERO((resresv->duration - used_amount));
+	return IF_NEG_THEN_ZERO((duration - used_amount));
```

Comment:
`IF_NEG_THEN_ZERO` will wrap argument with `()` so the inner pair of `()` is unnecessary.